### PR TITLE
[RUM 9603] Add RUM Auto test scenario to Benchmark app

### PR DESCRIPTION
**Pull request by sbarrio**
### What does this PR do?

This PR implements the benchmark test related to the RUM Auto Scenario.

### Motivation

We want to have a set of automatic benchmark tests run every time we merge code to develop.

#### iOS

Screenshot 2025-06-11 at 21 43 43 Screenshot 2025-06-11 at 21 43 50 Screenshot 2025-06-11 at 21 43 56 Screenshot 2025-06-11 at 21 44 05 Screenshot 2025-06-11 at 21 44 10 Screenshot 2025-06-11 at 21 44 30

#### Android
Screenshot 2025-06-11 at 21 44 58 Screenshot 2025-06-11 at 21 45 07 Screenshot 2025-06-11 at 21 45 14 Screenshot 2025-06-11 at 21 45 27 Screenshot 2025-06-11 at 21 45 35 Screenshot 2025-06-11 at 21 45 46 Screenshot 2025-06-11 at 21 45 52

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: benchmarks/src/scenario/RUM/Auto/rumAutoScenario.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+function RUMAutoScenario(props: RUMAutoScenarioProps): React.JSX.Element {
+    const instrumented = props.testConfig?.runType !== RunType.BASELINE;
+    const navigationRef = useNavigationContainerRef();
+    const viewPredicate: ViewNamePredicate = function customViewNamePredicate(route: Route<string, any | undefined>, trackedName: string) {
+        return `RN ${Platform.OS} Benchmark - RUM Auto - ${trackedName} / ${route.name}`;
+    };
+
+    const onDatadogInitialization = () => {
+        console.info("Datadog SDK initialized");
```
[the remark this answers is not in the document]
Comment (sbarrio) [reply benchmarks/src/scenario/RUM/Auto/rumAutoScenario.tsx:35]: We use this for testing purposes.

File: benchmarks/src/scenario/RUM/Auto/screens/episodes.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import RickMortyService from '../service/rickMorty';
+import { useNavigation } from '@react-navigation/native';
+import type { Episode } from '../service/types';
+
+function EpisodesScreen(): React.JSX.Element {    
+    const navigation = useNavigation<EpisodeDetailNavigationProp>();
+    const [nextPage, setNextPage] = useState<number | undefined>(undefined);
+    const [episodeData, setEpisodeData] = useState<Episode[]>([]);
+
+    useEffect(() => {
```
Comment (cdn34dd) [inline benchmarks/src/scenario/RUM/Auto/screens/episodes.tsx:27]: This logic is unsafe, if for instance the component unmounts before the promise resolves, you will get memory leaks. Or if you call getEpisodes multiple times because of pagination which you seem to be using here, there can happen the scenario where the next page can load before the first one leading to results being applied in incorrect order.

if you use something like react-query you don't have to think of these issues, you also get access to things like loading and error states, and retry handling.

**Review by cdn34dd — CHANGES_REQUESTED** [review]

File: benchmarks/src/scenario/RUM/Auto/screens/episodes.tsx (hunk 2 above)
Comment (sbarrio) [reply benchmarks/src/scenario/RUM/Auto/screens/episodes.tsx:27]: I agree, if this was a real consumer-facing app this logic would need a lot of refinement and edge case handling.

However, this is an app that will be used as a stress test scenario, run under an automated sequence of steps on synthetics and not used by real users, so edge cases are not actually important here. All we care about is measuring the cpu/memory/fps impact of our SDK when performing day to day tasks on a mobile app, and being able to monitor if any new features added to the SDK have a negative impact on those metrics.

That's why I've strived not to overcomplicate things across the board, nor to add extra dependencies into this app, in the same vein as the existing benchmark apps on the [Android ](https://github.com/DataDog/dd-sdk-android/blob/db25ca2e256d9cbe20de3345267b797275396cfe/sample/benchmark/src/main/java/com/datadog/benchmark/sample/ui/rumauto/screens/episodes/RumAutoEpisodesListViewModel.kt)and [iOS SDKs](https://github.com/DataDog/dd-sdk-ios/blob/0ece717cca152fcfddff9668f553c96a8e51c126/BenchmarkTests/Runner/Scenarios/RUM/RUMAuto/UI/Episodes/EpisodeDetailViewController.swift) already do.

**Review by maycmlee — COMMENTED** [review]
Nothing for docs to review

**Review by marco-saia-datadog — APPROVED** [review]

File: benchmarks/src/scenario/Logs/Custom/logsCustomScenario.tsx [hunk 3]
```diff
@@ -215,7 +224,9 @@ function LogsCustomScenario(props: LogsCustomScenarioProps): React.JSX.Element {
           </View>
           <View style={styles.separator} />
           <View style={styles.row}>
-            <Text style={styles.label}>{'Repeat logging'}</Text>
+            <Pressable onPress={onToggleIsRepeatLogging}>
+              <Text style={styles.label}>{'Repeat logging'}</Text>
+            </Pressable>
```
Comment (sbarrio) [inline benchmarks/src/scenario/Logs/Custom/logsCustomScenario.tsx:229]: Added these to make it easier to set during synthetics recordings.

File: benchmarks/src/scenario/Logs/Custom/logsCustomScenario.tsx [hunk 4]
```diff
@@ -136,6 +140,11 @@ function LogsCustomScenario(props: LogsCustomScenarioProps): React.JSX.Element {
         ...logOutputBuffer.current,
         `${new Date().toISOString()} - ${level} - ${size}: ${message}`,
       ];
+
+      if (logOutputBuffer.current.length > MAX_LOG_OUTPUT_ENTRIES) {
+        logOutputBuffer.current.shift();
+      };
```
Comment (sbarrio) [inline benchmarks/src/scenario/Logs/Custom/logsCustomScenario.tsx:146]: Decided to limit the logBuffer size so it does not get out of hand.

File: benchmarks/src/scenario/Logs/HeavyTraffic/imageList.tsx [hunk 5]
```diff
@@ -33,8 +39,9 @@ function ImageListScreen(props: ImageListProps): React.JSX.Element {
                 logger.current = DdLogs;
                 setReadyToRun(true);
             });
-        }
-
+        } else {
+            setReadyToRun(true);
```
Comment (sbarrio) [inline benchmarks/src/scenario/Logs/HeavyTraffic/imageList.tsx:43]: Bugfix: Non instrumented runs would not fetch anything 😓

File: benchmarks/src/testSetup/testUtils.ts [hunk 6]
```diff
@@ -19,8 +20,8 @@ import { Platform } from 'react-native';
 
 export const DEFAULT_ENV_TEST_CONFIG: TestConfig = {
     command: Command.Start,
-    scenario: Config.BENCH_SCENARIO,
-    runType: Config.BENCH_RUN_TYPE,
+    scenario: Config.BENCH_SCENARIO || undefined,
+    runType: Config.BENCH_RUN_TYPE || RunType.BASELINE,
```
Comment (sbarrio) [inline benchmarks/src/testSetup/testUtils.ts:24]: Non-baseline runs gets instrumented so we default to baseline in case this is undefined just to be sure.
